org.eclipse.xtext.builder.tests/src/org/eclipse/xtext/builder/impl/Bug486584Test.java

```diff
@@ -119,11 +173,30 @@ public void testFullBuildWhenClasspathChanged_3() throws CoreException, IOExcept
 		waitForBuild();
 		assertFalse(getEvents().isEmpty());
 		getEvents().clear();
+		
+		final CountDownLatch latch = new CountDownLatch(1);
+		ResourcesPlugin.getWorkspace().addResourceChangeListener(new IResourceChangeListener() {
```

Comment:
Nothing crucial, but these anonymous classes all look the same, so you can add a single implementation of the listener and pass the latch as a constructor argument, just to avoid code duplication